estela-api/api/serializers/auth.py

```diff
@@ -76,3 +75,30 @@ class UserProfileSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
         model = User
         fields = ["username", "email", "last_password_change"]
+
+
+class UserProfileUpdateSerializer(serializers.HyperlinkedModelSerializer):
+    last_password_change = serializers.ReadOnlyField(
+        source="userprofile.last_password_change", read_only=True
+    )
+
+    username = serializers.CharField(
+        validators=[
+            UniqueValidator(
+                queryset=User.objects.all(),
+                message="A user with that username already exists",
+            )
+        ]
+    )
+    email = serializers.CharField(
+        validators=[
+            UniqueValidator(
+                queryset=User.objects.all(),
+                message="A user with that email already exists",
+            )
+        ]
+    )
+
+    class Meta:
+        model = User
+        fields = ["username", "email", "password", "last_password_change"]
```

Comment:
We are duplicating the code here. `UserProfileUpdateSerializer` is very similar to `UserProfileSerializer`. If you want not to show the `password` field, you can set it to `write-only` so it is not returned in the API response.

Reply:
Thanks, deleted UserProfileUpdateSerializer and added password field with `write_only=True` to the UserProfileSerializer